Sources/_StringProcessing/RegexDSL/DSL.swift

```diff
@@ -39,6 +39,14 @@ extension Character: RegexProtocol {
   }
 }
 
+extension UnicodeScalar: RegexProtocol {
+  public typealias Match = Substring
+
+  public var regex: Regex<Match> {
+    .init(ast: atom(.scalar(self)))
```

Comment:
We need to stop constructing ASTs in the DSL. Can you use DSLTree, and if not, let me know what's missing?

Reply:
Switched this and the other primitives to use DSLTree 👍🏻